### ISSUE TITLE
SearchBar Component

### ISSUE THREAD

**Issue by jordanwillis** [saahil0826/jammming#4]
The `SearchBar` component looks pretty good.  Just a quick reminder to make sure you always initialize your state in the constructor.  This prevents unforeseen side effects and communicates what will be stored in the component state during its lifecycle.

Lastly, it looks like you are never initiating the search when the user clicks on the "search" button.

https://github.com/saahil0826/jammming/blob/master/src/Components/SearchBar/SearchBar.js#L26

You will need to define an `onClick` attribute on the search button like this.

``` javascript
<a onClick={this.search}>SEARCH</a>
```

**Comment by saahil0826** [comment]
ok made the change and what do you mean when you say initialize the state in the constructor? 
check line 12 in SearchBar is this fine? ok i did something like this   this.state = {
          term: '';
        };

which comes out as an error

**Comment by jordanwillis** [comment]
Sorry, I should have provided an example of what I meant by that!  Here is what is mean (note I've provided the full SearchBar constructor code).

``` javascript
constructor(props) {
  super(props);

  this.state = {
    term: ''
  };

  this.handleTermChange = this.handleTermChange.bind(this);
  this.search = this.search.bind(this);
}
```

So, I am initializing my state with a key to hold the search term (but of course the search term will be empty upon initialization).  I checked your constructor again (around line 12) but didn't see anything.  Did you forget to push some of your local changes?

https://github.com/saahil0826/jammming/blob/master/src/Components/SearchBar/SearchBar.js#L12

I'm not sure what error you might have received as this is how my implementation is and it all works fine.
